config/config.go

```diff
@@ -664,6 +671,43 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return checkOverflow(c.XXX, "kubernetes_sd_config")
 }
 
+// EC2SDConfig is the configuration for EC2 based service discovery.
+type EC2SDConfig struct {
+	Region          string                    `yaml:"region"`
+	Access          string                    `yaml:"access,omitempty"`
+	Secret          string                    `yaml:"secret,omitempty"`
+	RefreshInterval Duration                  `yaml:"refresh_interval,omitempty"`
+	Port            int                       `yaml:"port"`
+	Tags            map[string]string         `yaml:"tags,omitempty"`
```

Comment:
Every option below here should be done with relabelling, we need to dump the entire ec2 instance info into meta labels in a sane fashion.